### ISSUE TITLE
Line unfolding support not general enough

### ISSUE THREAD

**Issue by gracinet** [untitaker/rust-vobject#8]
First thank you for the lib
I was playing with it with calendaring purposes in mind, and got a parse error on an export from Google Calendar.

It seems that the parser expects line splits to occur in property values only (after the colon), whereas the RFC [makes it clear that it can occur anywhere](https://tools.ietf.org/html/rfc5545#section-3.1) and even between bytes of a single character.

With this test:

```rust
#[test]
fn test_icalendar_multline() {
    // Adapted from a very popular provider's export
    let event = parse_component(
        "BEGIN:VEVENT\n\
        ATTENDEE;CUTYPE=INDIVIDUAL;PARTSTAT=ACCEPTED;CN=Jo\n \
        hn Doe;X-NUM-GUESTS=0:mailto:jd@cal.test\n\
        SUMMARY:Important meeting\n\
        END:VEVENT\n").unwrap();

    assert_eq!(event.name, s!("VEVENT"));
    assert_eq!(event.single_prop("SUMMARY").unwrap().raw_value,
               s!("Important meeting"));
}
```
I get 

```
	thread 'test_icalendar_multline' panicked at 'called `Result::unwrap()` on an `Err` value: ParseError { desc: "Expected :, found \n" }', ../src/libcore/result.rs:741
```

with slight modifications, such as simply enclosing in `BEGIN:VCALENDAR` / `END:VCALENDAR` I get another error, the one I actually got on the real life export:

```
'called `Result::unwrap()` on an `Err` value: ParseError { desc: "No property name found." }'
```

I'll try and fix that if I can. That makes for an exercise in Rust.

**Comment by untitaker** [comment]
I see, I suspect we'll have to include some sort of preprocessor for unfolding. This is probably hard to do efficiently.

**Comment by gracinet** [comment]
I agree, the catch would be to avoid having to preprocess a huge calendar, with necessary copy etc.
in the meanwhile, I'm exploring having `consume_char()` use `peek()` to always look ahead. Not sure to be able to do that efficiently, but at least that's still streaming.

**Comment by untitaker** [comment]
Fixed by #9
